airflow/models/taskinstance.py

```diff
@@ -1839,6 +1839,36 @@ def key(self) -> TaskInstanceKey:
         """Returns a tuple that identifies the task instance uniquely."""
         return TaskInstanceKey(self.dag_id, self.task_id, self.run_id, self.try_number, self.map_index)
 
+    @staticmethod
+    @internal_api_call
+    @provide_session
+    def _set_state(ti, state, session=NEW_SESSION):
+        TI = TaskInstance
+        if not isinstance(ti, TaskInstance):
+            ti = (
+                session.query(TI)
```

Comment:
I dont know if it is really worth creating a variable for that

```suggestion
        TI = TaskInstance
        if not isinstance(ti, TaskInstance):
            ti = (
                session.query(TaskInstance)
```

Reply:
If `not isinstance(ti, TaskInstance)` is `True` then what is the type of `ti`? Should not we just pass `task_id`, `dag_id`, `run_id` and `map_index` as parameters? I think the goal is that all AIP-44 related APIs have primary keys as parameter which we then use to fetch the entity (task instance, dag_run, ...)

Reply:
TaskInstancePydantic in that case

Reply:
> Should not we just pass task_id, dag_id, run_id and map_index as parameters?

ok sure

Reply:
wait...
no so the reason for this is, if we already have a TI object there is no need to go fetch it from the DB.

wdyt?

Reply:
And if AIP-44 is enabled, what is sent through the wire? `TaskInstancePydantic` I guess, which is serializable. Makes sense to me

Reply:
ok